### PR TITLE
[DOCS]  Update `Process` / `Tech Direction` Sections for Tech Lead role

### PR DESCRIPTION
**Pull request by bashhack**
Motivation for changes:

* https://github.com/Mariana-Tek/engineering-ladder/issues/9

Changes:

* Add some consistency of syntax between [Engineering Manager](https://github.com/Mariana-Tek/engineering-ladder/blob/master/managers/engineering_manager.md) and [Tech Lead](https://github.com/Mariana-Tek/engineering-ladder/blob/master/roles/technical_lead.md)
* Expand on `Process` section
* Expand on `Tech Direction` section

### CODE REVIEW

File: roles/technical_lead.md [hunk 1]
```diff
@@ -22,33 +22,48 @@ It's the responsibility of the TL to ensure all of the following:
...
-* Manage technical WIP & help with/raise blockers
+* Manage technical work in progress (WIP)
+* Be aware of potential blockers and help to clear those that may arise
 * Ensure engineers have clear focus
 * Ensure engineers pause to ask questions as necessary
 
 #### Tech direction
 
 * Advise on which tech debt the team needs to address
+  * Actively engage in regular conversation to understand pain points for team
```
Comment (adsteel) [inline roles/technical_lead.md:41]: This feels borderline managerial, which I know many ICs who would be great tech leads specifically want to avoid. Would it make sense to keep responsibilities like this in the manager descriptions?

Comment (adsteel) [reply roles/technical_lead.md:41]: There are a few other items in this description that also feel managerial. So I'll just leave this comment here as a location to discuss them holistically.

e.g. 

- https://github.com/Mariana-Tek/engineering-ladder/pull/13/files#diff-a8a8ecea0546d071b105dd8464f7a305af3e49e3a139675784c7922c13d54ecdR66
- https://github.com/Mariana-Tek/engineering-ladder/pull/13/files#diff-a8a8ecea0546d071b105dd8464f7a305af3e49e3a139675784c7922c13d54ecdR47-R48

Comment (bashhack) [reply roles/technical_lead.md:41]: Ah, interesting points here!

I wonder if we're picking up on some ambiguity and blurring coming from my own experience serving both of these functions (manager + tech lead).

The items you've highlighted on the MT side definitely fall into the Tech Lead responsibilities more so than the Engineering Manager, historically. The Engineer Manager role at MT is very people-focused vs tech-focused, in many ways.

I think you're right to point out that on some teams backlog grooming or cross-team communication might be the responsibility of the EMs 🤔 Tricky! 

Curious what to hear from @jcarbaugh here too - what's the best way to capture the intended scope of ownership of these items? In the spirit of making edits that bring value to all teams 🤝 (not just MT), I'm open to all suggestions!

Comment (bashhack) [reply roles/technical_lead.md:41]: @tguidon - perhaps you have some thoughts on the above?

Comment (adsteel) [reply roles/technical_lead.md:41]: From a hiring/staffing perspective, I think the more we can separate technical leadership from people management the better. A lot of folks want to focus on technical challenges and will leave if given people management responsibilities. Definitely a bigger conversation to be had here around MT/TC alignment. After processing this for a while, I think this _might_ be a dealbreaker for me, which would necessitate TC having its own docs.

Comment (tguidon) [reply roles/technical_lead.md:41]: @bashhack This is tricky for sure! My current role is Team Lead, which I feel is a combination of a Tech Lead, Technical PM, and Engineering Manager, with a focus on moving towards EM. Maybe one issue is our Tech Lead role is written leaning too much towards this hybrid Team Lead role we have? The Team Lead role is not reflected on our ladder.

For the line in question, I do think it is a bit managerial, but I also think it's been an important skill for MT Tech Leads. I read this line as asking questions about the technical implementation of the project being worked on, rather than asking how a team member might be doing in their career journey. I believe a tech lead should be able to talk with their team and check in on how things are going. We could update the phrasing to better reflect that this isnt a weekly check in, but rather a project/technical level check in. Highlighting that they are not the manager, but rather a technical manager. Though in saying that I used the word manager 😅 

Looking at the other two points you mentioned @adsteel, those read as expectations of a Technical PM. MT does not have an official Technical PM role and I believe we have always lumped this in with the Tech Lead role. Definitely in my case and I think in Cathryn's as well back in the day.

I am curious how True Coach defines their Tech Lead role?

Comment (adsteel) [reply roles/technical_lead.md:41]: @tguidon At TC the tech lead role has never been this clearly defined, and each person puts their own spin on it, but my expectations have generally aligned with this document, minus the manager type points in discussion here.

Comment (adsteel) [reply roles/technical_lead.md:41]: Would it be worth getting together on a Zoom to talk about this?

Comment (tguidon) [reply roles/technical_lead.md:41]: @adsteel @bashhack I'm definitely down for a zoom sometime next week!

Comment (adsteel) [reply roles/technical_lead.md:41]: After talking with Jeremy about this last week, it sounds like his vision is for this role to merge with the Engineering Manager (and their be no "tech lead" title), which is a different direction than we're heading toward at TC. I think that means TC will end up with its own set of ladder docs, at least for a while. Though I'll be starting by taking 90% of the work we've done here. Maybe at some point in the future our structures will merge enough to merge the docs again.

tldr; TC and MT are diverging in their role definitions. Please continue this PR review without me.

Comment (bashhack) [reply roles/technical_lead.md:41]: Just saw this comment @adsteel - roger that! Thank you for the update and context there - and thank you for engaging in the task of helping improve these and other docs - they are stronger because of that input! 🙏

**Comment by bashhack** [comment]
FYI - @tguidon - slight change - with the `tech lead` to `team lead` change in flux, I added that as a topic for Boutique Management Sync (on agenda) and merging this as is temporarily. After that meeting, we can rename and keep --- or, migrate this content to the formal titles (EM + Staff + Senior + etc) if we deprecate the 'role' entry here.
